src/main/java/seedu/duke/data/timetable/RecurringEvent.java

```diff
@@ -107,6 +107,21 @@ public String toString() {
         return super.toString() + String.format(" (%s)", recurrenceType) + Formatter.LS + endRecurrenceString;
     }
 
+    @Override
+    public ArrayList<String> toStringArray() {
+        ArrayList<String> result = super.toStringArray();
+        String endRecurrenceString = "Until: ";
+        if (!endRecurrenceDate.equals(DEFAULT_END_RECURRENCE)) {
+            endRecurrenceString += endRecurrenceDate.toString();
+        } else {
+            endRecurrenceString += "Forever";
+        }
+        endRecurrenceString = " (" + endRecurrenceString + ")";
+        int lastIndex = result.size() - 1;
+        result.set(lastIndex, result.get(lastIndex) + endRecurrenceString);
```

Comment:
Same as the previous PR, can consider the following
```suggestion
        result.set(result.size()--, result.get(lastIndex) + endRecurrenceString);
```